src/meta/src/barrier/mod.rs

```diff
@@ -613,6 +620,14 @@ where
                     tracing::info!("Barrier manager is stopped");
                     return;
                 }
+                change_split_req = split_diff_rx.recv() => {
+                    let (command, notify_source_tx) = change_split_req.unwrap();
+                    // enqueue the source split config change command.
+                    self.scheduled_barriers.push(vec![(command,  once(Notifier {
+                        collected: Some(notify_source_tx),
+                        ..Default::default()
+                    }).collect())]).await;
+                }
```

Comment:
We may not add a special select arm for split change: `BarrierManager::run` acts like a consumer for the checkpoint queue and does not realize which specific type of command it runs. New commands should be enqueued externally.

Reply:
Well, this's a workaround to enqueue split config change command. Previously it's injected by calling `run_command` inside source manager. But for push down split assignments in recovery, the barrier manager will also need a reference of source manager, which will lead to circular references.
🤔, is there any other way we can cover these two scenarios and not lead to circular references?